pkg/collector/corechecks/system/cpu/cpu/cpu.go

```diff
@@ -83,36 +125,39 @@ func (c *Check) Run() error {
 		sender.Gauge("system.cpu.stolen", stolen*toPercent, "", nil)
 		sender.Gauge("system.cpu.guest", guest*toPercent, "", nil)
 	}
-
-	sender.Gauge("system.cpu.num_cores", c.nbCPU, "", nil)
-	sender.Commit()
-
 	c.lastNbCycle = nbCycle
 	c.lastTimes = t
 	return nil
 }
 
-// Configure the CPU check
-func (c *Check) Configure(senderManager sender.SenderManager, _ uint64, data integration.Data, initConfig integration.Data, source string) error {
-	err := c.CommonConfigure(senderManager, initConfig, data, source)
+func (c *Check) reportCpuMetricsPerCpu(sender sender.Sender) (err error) {
+	cpuTimes, err := getCpuTimes(true)
 	if err != nil {
+		log.Errorf("could not retrieve cpu stats: %s", err.Error())
```

Comment:
Instead of logging errors in the methods, could we just returns the errors from the main `Check.Run()` method and let the checkLogger log them?
https://github.com/DataDog/datadog-agent/blob/0f9693ff47d7ecc36a270ee2c056a991d25b0a07/pkg/collector/worker/worker.go#L189

Reply:
The `Check.Run()` method is called from many different places, not only from `checkLogger `